runelite-client/src/main/java/net/runelite/client/ClientSessionManager.java

```diff
@@ -100,15 +107,34 @@ private void ping()
 			log.warn(null, ex);
 		}
 
-		try
+
+		GameState gameState = client.getGameState();
+
+		if (gameState != GameState.LOGIN_SCREEN)

```

Comment:
(Apologies if I'm overstepping here, just looking around the codebase)

Could you not just set this to a variable and then use that on line 117/118 rather than having an if/else.